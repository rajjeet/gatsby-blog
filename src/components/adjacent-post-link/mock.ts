export const previous = {
  fields: { slug: '/previous/' },
  frontmatter: { title: 'Introducing the New React Concurrent Mode' },
};
export const next = {
  fields: { slug: '/next/' },
  frontmatter: { title: 'The Implications of React Concurrent Mode' },
};
