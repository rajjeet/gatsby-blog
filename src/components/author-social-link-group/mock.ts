import { TProps } from './types';

export const makeProps = (): TProps => ({
  className: '',
});

