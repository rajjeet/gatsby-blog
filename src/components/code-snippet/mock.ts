import { TProps } from './types';

export const makeProps = (): TProps => ({
  children: `
const mainReducer = (state, action) => {
  console.log('Before', state);
  console.log('Action', action);
  let result = {
    count: countReducer(state.count, action)      
  };
  console.log('After', result);
  return result
};      
`,
  fileName: 'index.js',
});
