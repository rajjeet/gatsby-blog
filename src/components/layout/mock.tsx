import React from 'react';
import { TProps } from './types';
import { createMockGatsbyImageSharpFluid } from '../../utils/testing';
import { siteMetadata } from '../../../gatsby-config';

export const makeProps = (): TProps => ({
  children: <div>Sample child</div>,
  data: {
    file: createMockGatsbyImageSharpFluid.file,
    site: siteMetadata,
  },
});

