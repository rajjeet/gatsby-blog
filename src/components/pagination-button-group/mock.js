export const makeProps = () => ({
  className: {}, currentPage: 1, numOfPages: 5, paginationSlug: '',
});
