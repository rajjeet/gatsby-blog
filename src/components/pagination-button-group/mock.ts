export const makeProps = ({ currentPage = 1, numOfPages = 5 } = {}): TProps => ({
  className: '', currentPage, numOfPages, paginationSlug: '',
});

export type TProps = {
  className: string;
  currentPage: number;
  numOfPages: number;
  paginationSlug: string;
};
