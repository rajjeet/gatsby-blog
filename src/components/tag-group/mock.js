import { createMockGroups } from '../../utils/testing';

export const makeProps = () => ({
  className: {}, tags: createMockGroups.categoryGrouping.group, getSlug: (s) => s,
});
