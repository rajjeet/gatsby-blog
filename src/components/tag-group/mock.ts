import { mockGroups } from '../../fixtures/tags';
import { TProps } from './types';

export const makeProps = (): TProps => ({
  tags: mockGroups.categoryGrouping.group,
  getSlug: (s): string => s,
});

