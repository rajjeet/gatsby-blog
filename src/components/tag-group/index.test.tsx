import React from 'react';
import { render } from '@testing-library/react';
import { TagGroup } from './index';
import { makeProps } from './mock';

describe('<TagGroup />', () => {
  it('should render', () => {
    const { asFragment } = render(<TagGroup {...makeProps()} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should link for each tag group', () => {
    const { queryAllByRole } = render(<TagGroup {...makeProps()} />);
    expect(queryAllByRole('link')).toHaveLength(2);
  });

  it('should the name of each tag group', () => {
    const { getByText } = render(<TagGroup {...makeProps()} />);
    expect(getByText('Experience')).toBeDefined();
    expect(getByText('Guide')).toBeDefined();
  });

  it('should the count of each tag group', () => {
    const { getByText } = render(<TagGroup {...makeProps()} />);
    expect(getByText('1')).toBeDefined();
    expect(getByText('2')).toBeDefined();
  });
});
