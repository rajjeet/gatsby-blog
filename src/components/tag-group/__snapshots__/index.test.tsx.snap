// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TagGroup /> should render 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 {
  padding: .2rem .5rem;
  margin-right: .5rem;
  margin-top: .25rem;
  font-weight: bolder;
  font-size: .8rem;
}

.c1:hover {
  cursor: pointer;
}

.c2 {
  margin-left: 1em;
}

<div
    class="c0"
  >
    <span
      class="c1"
      role="link"
      tabindex="0"
    >
      Experience
      <span
        class="c2"
      >
        1
      </span>
    </span>
    <span
      class="c1"
      role="link"
      tabindex="0"
    >
      Guide
      <span
        class="c2"
      >
        2
      </span>
    </span>
  </div>
</DocumentFragment>
`;
